### PR TITLE
Quick fix #617

### LINKED ISSUE #617: Throws on iOS 10
Including `webcomponents.js` v0.7.22 in Safari on iOS 10.0.1 throws

```
TypeError: undefined is not an object (evaluating 'descriptor.value')
installProperty — webcomponents.js:235
registerInstanceProperties — webcomponents.js:182
registerInternal — webcomponents.js:269
register — webcomponents.js:260
(anonymous function) — webcomponents.js:4365
global code — webcomponents.js:4367
```

The part in question is
```js
var descriptor = getDescriptor(source, name);
var getter, setter;
if (typeof descriptor.value === "function") {
```

and the error happens when `source` is the window object and `name` is `'showModalDialog'` (returns `undefined`).

It can be quick-fixed by changing `getDescriptor` in `ShadowDOM/wrappers.js` to:

```js
  function getDescriptor(source, name) {
    try {
      return Object.getOwnPropertyDescriptor(source, name) || dummyDescriptor;
    } catch (ex) {
      // JSC and V8 both use data properties instead of accessors which can
      // cause getting the property desciptor to throw an exception.
      // https://bugs.webkit.org/show_bug.cgi?id=49739
      return dummyDescriptor;
    }
  }
```

### PR DESCRIPTION
**Pull request by qwtel**

### CODE REVIEW

File: src/ShadowDOM/wrappers.js [hunk 1]
```diff
@@ -182,7 +182,7 @@ window.ShadowDOMPolyfill = {};
 
   function getDescriptor(source, name) {
     try {
-      return Object.getOwnPropertyDescriptor(source, name);
+      return Object.getOwnPropertyDescriptor(source, name) || dummyDescriptor;
```
Comment (addyosmani) [inline src/ShadowDOM/wrappers.js]: My expectation is this should throw an exception which we catch in the guard lower down. As the errors happens due to certain conditions with `window` and `showModalDialog`, is there a reason we wouldn't want to check for source === window && name === 'showModalDialog' and only early return the extra dummyDescriptor there?

**Comment by tschwartz** [comment]
@qwtel Thoughts?

**Comment by qwtel** [comment]
My assumption was that `showModalDialog` could be the first of many such errors. Turns out it's the only one. I've updated the PR.

File: src/ShadowDOM/wrappers.js (hunk 1 above)
Comment (qwtel) [inline src/ShadowDOM/wrappers.js]: My assumption was that showModalDialog could be the first of many such errors. Turns out it's the only one. I've updated the PR.

**Comment by phpthug** [comment]
@addyosmani @ebidel Any additional feedback for @qwtel? As iOS10 becomes more widely used, this is going to start impacting more users.

**Comment by mattppp** [comment]
Any update on this?

**Comment by ArturKlajnerok** [comment]
Hey, @addyosmani @azakus @sorvell is there anything we could help / change to get this fix out ?
Since iOS 10 seems to have more than 40% market share this quite an issue  😸

**Comment by jbwyme** [comment]
Is anyone actually actively maintaining this lib?

**Comment by zanona** [comment]
Crazy isn't it. ¬¬

**Comment by phpthug** [comment]
Any idea when this will make it into a tagged release?

**Comment by samyue** [comment]
I have got this iphone ios10 issue as well. are there any ways for us to work around this issue before official release?

**Comment by jannakha** [comment]
I had to use issue #617 fix and move ALL webcomponentsjs/* files into S3 to make it work, because webcomponents.js is not min'ed and not still pulls all refs dynamically 

@addyosmani please release ASAP, production is suffering

**Comment by samyue** [comment]
@jannakha yeah, using the #617 fix to modify the getDescriptor function is working

**Pull request merged**